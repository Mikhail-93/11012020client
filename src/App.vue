<template>
    <v-app id="inspire">
        <v-navigation-drawer v-model="drawer" app>
            <v-list dense>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-home</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Home</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-contact-mail</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Contact</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
            </v-list>
        </v-navigation-drawer>

        <v-app-bar
                app
                color="indigo"
                dark
        >
            <v-app-bar-nav-icon @click.stop="drawer = !drawer"/>
            <v-toolbar-title>Application</v-toolbar-title>
        </v-app-bar>

        <v-content>
            <v-container class="fill-height pa-0" fluid>
                <LMap ref="map" style="z-index: 0">
                    <l-tile-layer
                            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                            attribution='Map data: &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)'
                    >
                    </l-tile-layer>
                    <!--                    <l-control-attribution position="bottomright" prefix="leaflet OpenStreetMap Contributions" >-->

                </LMap>
                <!--                class="align-stretch"-->
            </v-container>
        </v-content>
    </v-app>
</template>

<script>
    import {LMap, LTileLayer, LMarker, LControlAttribution} from 'vue2-leaflet';

    export default {
        components: {
            LMap,
            LTileLayer,
            LMarker,
            LControlAttribution
        },
        props: {
            source: String,
        },
        data: () => ({
            drawer: null,
        }),
        mounted() {
            this.$nextTick(() => {
                // this.map=
                this.$refs.map.mapObject.setView([51.505, -0.09], 13)

            })
        }
    }
</script>