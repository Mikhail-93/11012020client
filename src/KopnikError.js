class KopnikError extends Error {
    constructor(message, code, base) {
        super(message)
        this.code = code
        this.base= base
    }
}
class KopnikApiError extends KopnikError {
    constructor(message, code, url, base) {
        super(message, code, base)
        this.url = url
    }
}

export {KopnikError, KopnikApiError}
