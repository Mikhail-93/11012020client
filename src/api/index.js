import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
