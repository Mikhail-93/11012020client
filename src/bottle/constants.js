/**
 * Created by alexey2baranov on 8/20/16.
 *
 * домены и редикт урлы
 * https://vk.com/apps?act=manage
 * дебаг на телефоне
 * https://developers.google.com/web/tools/chrome-devtools/remote-debugging?utm_source=dcc&utm_medium=redirect&utm_campaign=2016q3
 */
let constants = {
  "development": {
    sw: {
      delay: 15000,
    },
    messenger: {
      // идентификатор нашего приложения ВК
      // clientId: 7210289,
      // страница, на которую нужно перейти для аутентификации
      loginUrl: 'http://localhost:8081/connect/vkontakte',
      // с мобилки
      // loginUrl: 'http://192.168.43.84:8081/connect/vkontakte'
    },
    api: {
      path: "http://localhost:8081/api"
      // с мобилки
      // path: "http://192.168.43.84:8081/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  },
  "test": {
    sw: {
      delay: Number.MAX_SAFE_INTEGER,
    },
    messenger: {
      // clientId: 7210289,
      loginUrl: 'http://localhost:8082/connect/vkontakte',
      // redirectUrl: 'http://localhost:8082/login/check-vk'
    },
    api: {
      path: "http://localhost:8082/api"
    },
    di: {
      fetch: false,
      cookie: true
    }
  },
  "staging": {
    sw: {
      delay: 20000,
    },
    messenger: {
      loginUrl: 'https://staging.kopnik.org/connect/vkontakte',
      // clientId: 7210289,
      // redirectUrl: 'https://staging.kopnik.org/login/check-vk'
    },
    api: {
      path: "https://staging.kopnik.org/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  }
}

export default constants
