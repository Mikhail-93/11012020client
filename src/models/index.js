/**
 * Created by alexey2baranov on 8/20/16.
 */

import AbstractSync from "./AbstractSync" //для того чтобы правильно работали рекурсивные ссылки модулей
// import Registration from "./model/Registration"
import Kopnik from "./Kopnik"
// import Kopa from "./model/Kopa"
// import File from "./model/File"

export {AbstractSync, Kopnik}