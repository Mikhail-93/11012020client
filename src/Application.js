import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
