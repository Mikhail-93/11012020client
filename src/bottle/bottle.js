import Bottle from "bottlejs"
import _ from "lodash"
// import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../application/Application"
import constants from './constants'
import api from "../api";
import CookieService from "./CookieService"
import MK from "../mk/mk";
import LocaleManager from "../locales/LocaleManager";
import ru from '../locales/ru'
import en from '../locales/en'

Bottle.config.strict = true
const bottle = new Bottle()

bottle.factory('logger', function loggerFactory() {
    //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
    // LoglevelPluginPrefix.reg(loglevel)
    // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
    loglevel.setLevel(loglevel.levels.DEBUG)
    if (process.env.NODE_ENV == "test") {
        loglevel.setLevel(loglevel.levels.DEBUG)
        // LoglevelPluginToString.apply(loglevel, {})
    }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
    return loglevel
})

bottle.constant('messages', {
    ru,
    en,
})

bottle.factory('env', function env() {
    return process.env.NODE_ENV || 'development'
})
bottle.service('localeManager', function localeManager() {
    const localeManager = new LocaleManager()
    localeManager.currentLocale = 'ru'//container.env === 'development' ? 'en' : 'ru'
    return localeManager
})
bottle.service('cookieService', CookieService, 'constants')
bottle.factory('api', function apiFactory(container) {
    if (!container.constants.di.fetch) {
        throw new Error('not supported')
    }
    return api
})
bottle.factory('VK', function vkFactory(container) {
    return process.env.NODE_ENV === 'test' ? MK : global.VK
})
bottle.factory('constants', function constantsFactory() {
    if (!process.env.NODE_ENV) {
        throw new Error("NODE_ENV is not defined")
    }

    let local = {}//require("./local.js")
    let result = _.merge({}, constants, local)[process.env.NODE_ENV]

    return result
})
if (!Application) {
    throw new Error("Application is undefined in bottle")
}
bottle.service('application', Application, 'logger')

/**
 * @callback fetch
 * @param {string} url
 * @param {Object?} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} api
 * @property {MK.class} VK
 * @property {CookieService} cookieService
 * @property {Object} constants
 * @property {Object} constants.api
 * @property {String} constants.api.path
 * @property {Object} constants.di
 * @property {Boolean} constants.di.fetch
 * @property {Number} constants.di.cookie
 * @property {Number} constants.messenger.clientId
 * @property {String} constants.messenger.loginUrl
 * @property {String} constants.messenger.redirectUrl
 * @property {String} constants.sw
 * @property {String} constants.sw.delay
 * @property {LocaleManager} localeManager
 * @property {string} env
 * @property {{ru, en}} messages
 */
const container = bottle.container
export {bottle, container}
