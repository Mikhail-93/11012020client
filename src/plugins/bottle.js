import Bottle from "bottlejs"
import _ from "lodash"
import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../Application"
import config from '../../config'
import fetchApi from "../bottle/fetchApi";
import fetchApiMock from "../bottle/fetchApi.mock";

Bottle.config.strict = true
const bottle = new Bottle()

/**
 * Настраивает контейнер
 * Для тестов необходим сброс и перенастройка чтобы соблюсти изолированность тестов
 *
 * @param {Object} options параметры по умолчанию соответствуют наиболее приемлемым для окружения значениям окружению
 * @param {Number=} options.cookie
 * @param {Boolean=} options.fetch
 */
Bottle.prototype.setup = function (options) {
    bottle.factory('defaultFetchApiOptions', function defaultFetchApiOptionsFactory(container) {
        const cookies = {
            1: 'PHPSESSID=user-1',
            2: 'PHPSESSID=eqlae6a1grpvcirpemfvebf806',
            3: 'PHPSESSID=user-3',
            4: 'PHPSESSID=user-4',
            5: 'PHPSESSID=user-5',
            6: 'PHPSESSID=user-6',
            7: 'PHPSESSID=user-7',
            8: 'PHPSESSID=user-8',
            9: 'PHPSESSID=user-9',
            10: 'PHPSESSID=user-10',
        }
        const result = {}
        if (container.config.di.cookie) {
            result.headers = {
                cookie: cookies[container.config.di.cookie]
            }
        }
        // console.log(result)
        return result
    })
    bottle.factory('fetchApi', function fetchApiFactory(container) {
        return container.config.di.fetch ? fetchApi : fetchApiMock
    })
    bottle.factory('config', function configFactory() {
        if (!process.env.NODE_ENV) {
            throw new Error("NODE_ENV is not defined");
        }

        let local = {}//require("./local.js")
        let result = _.merge({}, config, local)[process.env.NODE_ENV]

        return result
    })
    if (!Application){
        throw new Error("Application is undefined in bottle")
    }
    bottle.service('application', Application, 'logger')
    bottle.factory('logger', function loggerFactory() {
        //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
        // LoglevelPluginPrefix.reg(loglevel)
        // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
        loglevel.setLevel(loglevel.levels.DEBUG)
        if (process.env.NODE_ENV == "test") {
            loglevel.setLevel(loglevel.levels.WARN)
            // LoglevelPluginToString.apply(loglevel, {})
        }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
        return loglevel
    })
    Object.assign(this.container.config.di, options)
}

bottle.setup()

/**
 * @callback fetch
 * @param {string} url
 * @param {Object} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} fetchApi
 * @property {Object} defaultFetchApiOptions
 * @property {Object} defaultFetchApiOptions.headers
 * @property {Object} defaultFetchApiOptions.headers.cookie
 * @property {Object} config
 */
const container = bottle.container
export {bottle, container}
