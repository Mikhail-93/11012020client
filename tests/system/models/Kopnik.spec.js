import {KopnikApiError} from "../../../src/KopnikError";
import {AbstractSync, Kopnik} from "../../../src/models";
import {bottle, container} from "../../../src/bottle/bottle";

container.config.di.fetch = true

describe('system models Kopnik', () => {
    let kopnik

    beforeEach(() => {
        bottle.resetProviders(['cookieService'])
        AbstractSync.clearCache()
    })

    it('locale present', async () => {
        await login(1)
        const kopnik=  await Kopnik.get(1)
        expect(kopnik.locale).toBe('ru')
    })
    it('location is object', async () => {
        await login(1)
        const kopnik=  await Kopnik.get(1)
        expect(kopnik.location).toHaveProperty('lat')
        expect(kopnik.location).toHaveProperty('lng')
    })
    it('updateLocale', async () => {
        await login(1)
        const kopnik=  await Kopnik.get(1)
        await kopnik.updateLocale('ru')
    })
    it('isMessagesFromGroupAllowed', async () => {
        await login(1)
        const kopnik=  await Kopnik.get(1)
        const result= await kopnik.isMessagesFromGroupAllowed()
        expect(result).toBe(true)
    })
    it('getWitnessRequests have passport', async () => {
        await login(1)
        const kopnik= Kopnik.getReference(1)
        await kopnik.reloadWitnessRequests()
        expect(kopnik.witnessRequests).toBeInstanceOf(Array)
        expect(kopnik.witnessRequests[0]).toHaveProperty('passport')
        expect(kopnik.witnessRequests[0].passport).toBe(5220)
    })
    it('get havn\'t passport', async () => {
        await login(1)
        const kopnik=  await Kopnik.get(3)
        expect(kopnik.passport).toBeFalsy()
    })
})
