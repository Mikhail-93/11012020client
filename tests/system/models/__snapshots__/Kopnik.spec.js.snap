// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`system models Kopnik confirmed update kopnik1@users/update 1`] = `
Array [
  Object {
    "birthyear": 1985,
    "firstName": "Павел",
    "foreman_id": null,
    "id": 1,
    "lastName": "Сысоев",
    "location": Array [
      47.042697,
      147.94418,
    ],
    "passport": 9991,
    "patronymic": "Якушев",
    "photo": "@todo",
    "smallPhoto": "@todo",
    "status": 1,
    "witness_id": 1,
  },
]
`;
